src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -3294,21 +3294,17 @@ void c_typecheck_baset::typecheck_function_call_arguments(
     }
     else if(i<parameter_types.size())
     {
-      const code_typet::parametert &parameter_type=
```

Comment:
:pick: Could you separate this refactor out into a separate commit from the using of `exprt::is_boolean()`? This would have made it easier to review and it could make it easier to bisect later, if we find problems have been introduced.

Reply:
I have now dropped this refactor from this PR.